api/pkg/cli/cmd/info/info.go

```diff
@@ -0,0 +1,192 @@
+
+{{- else -}}
+
+{{ icon "name" }}Name: {{ .ResVersion.Resource.Name }} 
+{{ $n := .ResVersion.DisplayName }}{{ if ne (default $n "") "" }}
+{{ icon "displayName" }}Display Name: {{ $n }}
+{{ end }}
+{{ icon "version" }}Version: {{ .ResVersion.Version }} 
+
+{{ icon "description" }}Description: {{ formatDesc .ResVersion.Description 80 16 }}
+
+{{ icon "minPipelineVersion" }}Minimum Pipeline Version: {{ .ResVersion.MinPipelinesVersion }}
+
+{{ icon "rating" }}Rating: {{ .ResVersion.Resource.Rating }}
+
+{{ $t := len .ResVersion.Resource.Tags }}{{ if ne $t 0 }}
+{{- icon "tags" }}Tags
+ {{- range $p := .ResVersion.Resource.Tags }}
+  {{ icon "bullet" }}{{ $p.Name }}
+ {{- end }}
+{{- end }}
+
+{{ icon "install" }}Install Command:
+  tkn hub install {{ lower .ResVersion.Resource.Kind }} {{ .ResVersion.Resource.Name }} --version {{ .ResVersion.Version }}
+  {{- if ne (default .ResVersion.Resource.Catalog.Name "") "tekton" }} --from {{ .ResVersion.Resource.Catalog.Name }}{{ end }}
+
+{{- end }}
+`
+
+var (
+	funcMap = template.FuncMap{
+		"icon":       formatter.Icon,
+		"formatDesc": formatter.WrapText,
+		"default":    formatter.DefaultValue,
+		"lower":      strings.ToLower,
+	}
+	tmpl = template.Must(template.New("Resource Info").Funcs(funcMap).Parse(resTemplate))
+)
+
+type options struct {
+	cli     app.CLI
+	from    string
+	version string
+	kind    string
+	args    []string
+}
+
+func Command(cli app.CLI) *cobra.Command {
+
+	opts := &options{cli: cli}
+
+	cmd := &cobra.Command{
+		Use:   "info",
+		Short: "Describe info of resource by its name, kind, catalog, and version",
+		Long:  ``,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		commandForKind("task", opts),
+	)
+
+	cmd.PersistentFlags().StringVar(&opts.from, "from", "tekton", "Name of Catalog to which resource belongs.")
+	cmd.PersistentFlags().StringVar(&opts.version, "version", "", "Version of Resource")
+
+	return cmd
+}
+
+// commandForKind creates a cobra.Command that when run sets
+// opts.Kind and opts.Args and invokes opts.run
+func commandForKind(kind string, opts *options) *cobra.Command {
+
+	return &cobra.Command{
+		Use:          kind,
+		Short:        "Describe info of " + kind + " by name, catalog and version",
+		Long:         ``,
+		SilenceUsage: true,
+		Annotations: map[string]string{
+			"commandType": "main",
+		},
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.kind = kind
+			opts.args = args
+			return opts.run()
+		},
+	}
+}
+
+func (opts *options) run() error {
+
+	if err := opts.validate(); err != nil {
+		return err
+	}
+
+	hubClient := opts.cli.Hub()
+	res := hubClient.GetResource(hub.ResourceOption{
+		Name:    opts.name(),
+		Catalog: opts.from,
+		Kind:    opts.kind,
+		Version: opts.version,
+	})
+
+	resource, err := res.Resource()
+	if err != nil {
+		return err
+	}
+
+	out := opts.cli.Stream().Out
+
+	if opts.version != "" {
```

Comment:
Is not better to just return the resource based on the version asked, we can remove the if/else thing from the template then

Reply:
updated